SU-HSDO/stanford_mrc
# Mrcd8 163 multiple file upload

# READY FOR REVIEW

# Summary
- Added page to bulk upload multiple files in one drag and drop
- Added media delete action & bulk operations on that action
- Replace core /admin/content/media page with our own
- Added entity_usage module
- Added entity usage report to the admin/content/media page with links to the referencing entities.
- Fixed embed video tab in media browser.

# Needed By (Date)
- Nov 7th

# Urgency
- High

# Steps to Test

1. Checkout latest `develop` branch of [mrc_blt](https://github.com/su-hsdo/mrc_blt)
2. run `composer update --prefer-source`
3. Checkout this branch of `stanford_mrc`
4. run `drush updb -y; drush cr`
5. Navigate to /admin/content/media page
6. Click the "add Media" link at the top
7. Click the "Bulk Upload" link. this should bring you to the drag and drop similar to the wysiwyg.
8. Drag some files in.
9. Ensure they upload and your are presented with another form for each file to edit the title, alt text etc.
10. Go to create/edit a node.
11. embed the media you just uploaded and ensure it looks as expected.
12. Embed a video while you're there to see if it is fixed.
13. After saving the content with the media items, go back to the /admin/content/media page
14. Ensure the usage has incremented for the media you embeded.
15. click on the "Used In" for that item.
16. validate you see a summary of where the media item is embed and you can click to that entity.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
